### ISSUE TITLE
Repo Mirroring settings page doesn't report password expiry for p4-fusion-cloned Perforce depots

### ISSUE THREAD

**Issue by twarit-waikar** [sourcegraph/sourcegraph#44031]
- **Sourcegraph version: 3.42** 
- **Platform information: Alpine (Docker)** 

#### Steps to reproduce:

1. Clone a Perforce depot using a valid p4-fusion config i.e. with `fusionClient.enabled = true`
2. Manually expire the Perforce user password by separately running `p4 admin resetpassword -u <perforce_username>`
3. Go to Sourcegraph repo's Mirroring page, click on 'Refresh Now'
4. See that Sourcegraph does not report an expired password when trying to fetch the depot again.

The workaround here is for the admin to log into the server, meanwhile click on 'Refresh Now' from the UI, run ```cat /proc/`pgrep p4-fusion`/fd/1``` from the shell and check for any error messages. If p4-fusion doesn't seem to actually run, it is probably returning too fast after detecting the error.

But since p4-fusion versions <=1.11 do not report errors like this as exit codes, Sourcegraph is unaware that something has gone wrong. This has been fixed with [p4-fusion v1.12.0](https://github.com/salesforce/p4-fusion/releases/tag/v1.12)

#### Expected behavior:
See the proper error message saying that the cloning Perforce user's password has expired in the Sourcegraph UI. And then see the error get resolved on its own by resetting the Perforce password.

#### Actual behavior:
No errors are shown.

If would like immediate help on this, please email support@sourcegraph.com (you can still create the issue, but there are not [SLAs](https://about.sourcegraph.com/support/) on issues like there are for support requests).

/cc @jplahn @ryphil

**Comment by peterguy** [comment]
I think this can be closed. Errors like this are surfaced reliably as of #43855

**Comment by peterguy** [comment]
estimate 1 day for research

**Comment by indradhanush** [comment]
I tested this with p4-fusion `v1.12.0` and here's what I found:


1. Completed a sync successfully with a valid config (see screenshot)
2. Reset password for user `dhanush` and ran a sync - the expected error is seen in the code host sync details
3. Set a new password for user `dhanush` and did not update the `ticket` in the field `p4.passwd` and it showed a new error
4. Generated a new ticket by running `P4USER=dhanusn p4 login -a` and updated the code host config, the sync ran successfully again

<img width="1070" alt="image" src="https://user-images.githubusercontent.com/2682729/233349271-d277b5ad-a299-44be-ae52-380188a0e924.png">


As mentioned in the issue description - this seems to be related with an older version of `p4-fusion` and now with `v.12.0` we are able to see the errors.
